src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,42 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a Person's contact in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValid(String, String)}
+ */
+public class Contact {
+    public String value;
+    protected boolean isPrivate;
+
+    public Contact(boolean isPrivate) {
+        this.isPrivate = isPrivate;
+    }
+
+    /**
+     * Returns true if a given string is a valid person phone/email/address.
+     */
+    public static boolean isValid(String test, String validationRegex) {
+        return test.matches(validationRegex);
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
```

Comment:
This could be a potential problem. You may end up equating email is a phone number. Think why?